File: src/Omnia.ProcessManagement.Models/Processes/Process.cs
using Newtonsoft.Json.Linq;
using Omnia.Fx.Models.Language;
using Omnia.ProcessManagement.Models.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.Processes
{
    public class Process
    {
        public Guid Id { get; set; }
        public Guid OPMProcessId { get; set; }
        public RootProcessStep RootProcessStep { get; set; }
        public string CheckedOutBy { get; set; }
        public ProcessVersionType VersionType { get; set; }
        public Guid TeamAppId { get; set; }
        public ProcessWorkingStatus ProcessWorkingStatus { get; set; }
        public Guid SecurityResourceId { get; set; }
        public string CreatedBy { get; set; }
        public string ModifiedBy { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset ModifiedAt { get; set; }
        public DateTimeOffset? PublishedAt { get; set; }
    }
}

File: src/Omnia.ProcessManagement.Web/client/fx/models/data/processes/Process.ts
import { GuidValue } from '@omnia/fx-models';
import { RootProcessStep } from './RootProcessStep';
import { ProcessVersionType, ProcessWorkingStatus } from '..';

export interface Process {
    id: GuidValue;
    opmProcessId: GuidValue;
    rootProcessStep: RootProcessStep;
    versionType: ProcessVersionType;
    teamAppId: GuidValue;
    webId: GuidValue;
    processWorkingStatus: ProcessWorkingStatus;
    checkedOutBy?: string;
    createdBy: string;
    modifiedBy: string;
    createAt: Date;
    modifiedAt: Date;
    publishedAt?: Date;
    publishedBy?: string;

    //client-side
    modifiedByName?: string;
    checkedOutByName?: string;
}